tests/mesh/mesh_smoother_test.C

```diff
@@ -345,6 +351,54 @@ public:
       }
   }
 
+  // Helper function to determine how many dimensions of a given point lie at
+  // the center and faces of a sub-cube
+  std::pair<unsigned int, unsigned int>
+  numCenteredAndFacedDimensions(const Point & point, const Real & side_length, const Real & tol)
+  {
+    const auto half_length = 0.5 * side_length;
+    unsigned int num_centered = 0;
+    unsigned int num_face = 0;
+    for (const auto d : make_range(3))
+      {
+        const auto num_half_lengths = point(d) / half_length;
+        if (std::abs(num_half_lengths - std::round(num_half_lengths)) > tol)
+          // This coordinante does not occur at a sub-cube face or center
+          continue;
+
+        if ((unsigned int)std::round(num_half_lengths) % 2)
```

Comment:
Don't let John catch you writing a C-style cast!

Also, do we want to `std::round` and then cast?  Wouldn't `std::lround` be more straightforward?